pkg/server/endpoints/node/handler.go

```diff
@@ -523,6 +523,28 @@ func (h *Handler) PushJWTKeyUpstream(ctx context.Context, req *node.PushJWTKeyUp
 	}, nil
 }
 
+func (h *Handler) FetchBundle(ctx context.Context, req *node.FetchBundleRequest) (_ *node.FetchBundleResponse, err error) {
+	counter := telemetry_server.StartNodeAPIFetchBundleCall(h.c.Metrics)
+	defer counter.Done(&err)
+	log := h.c.Log.WithField(telemetry.Method, telemetry.FetchBundle)
+
+	_, ok := getPeerCertificate(ctx)
```

Comment:
nit: I don't think this check is necessary since AuthorizeCall should ensure there is a client certificate.